src/main/java/com/bp3/camunda/camunda7/C7RESTConnector.java

```diff
@@ -79,17 +84,57 @@ public void execute(ExternalTask externalTask, ExternalTaskService externalTaskS
         setPayload(request, httpMethod, httpPayload);
 
         // call the REST service
-        HttpResponse response = request.execute();
+        try {
+            HttpResponse response = request.execute();
+
+            // set the output variable
+            log.debug("RESPONSE: {}", response.getResponse());
+            VariableMap variables = Variables.createVariables();
+            if (outputVariableName != null) {
+                variables.putValue(outputVariableName, response.getResponse());
+            }
 
-        // set the output variable
-        log.debug("RESPONSE: {}", response.getResponse());
-        VariableMap variables = Variables.createVariables();
-        if (outputVariableName != null) {
-            variables.putValue(outputVariableName, response.getResponse());
-        }
+            // complete the external task
+            externalTaskService.complete(externalTask, variables);
+        } catch(ConnectorException e) {
+            log.error("CONNECTOR_ERROR: {}", e.getLocalizedMessage(), e.fillInStackTrace());
+
+            String errorHandlingMethod = (String) getVariable(externalTask, PARAM_ERROR_HANDLING_METHOD, String.class);
+
+            if (errorHandlingMethod != null) {
+                log.debug("CONNECTOR_ERROR: Handling as '{}'", errorHandlingMethod);
+
+                int totalRetries = 0; // Default to zero if not set
```

Comment:
These can be moved further down to the `case "Failure"` lamda where they are used

Reply:
Good point, now moved.